contracts/Vault.vy

```diff
@@ -1585,16 +1585,32 @@ def _assessFees(strategy: address, gain: uint256) -> uint256:
     if gain == 0:
         # NOTE: The fees are not charged if there hasn't been any gains reported
         return 0
+    
+    undelegatedAssets: uint256 = self.strategies[strategy].totalDebt - Strategy(strategy).delegatedAssets()
+
+    performance_fee: uint256 = 0
+    management_fee: uint256 = 0
+
+    if undelegatedAssets > 0:
+        management_fee = (
+            (
+                undelegatedAssets
+                * duration 
+                * self.managementFee
+            )
+            / MAX_BPS
+            / SECS_PER_YEAR
+        )
 
-    management_fee: uint256 = (
-        (
-            (self.strategies[strategy].totalDebt - Strategy(strategy).delegatedAssets())
-            * duration 
-            * self.managementFee
+         # NOTE: Unlikely to throw unless strategy reports >1e72 harvest profit
+        performance_fee = (
```

Comment:
If governance gets 0 fee for fully delegated strategy, but strategist gets the full fee, it's kind of a weird situation.

It might be better to have more control over this value by moving the performance fee to governance to being independently set per-strategy

Reply:
My preference would be that all fees (gov perf, strategist perf, and mgmt) get handled using the delegation logic. To me it is the more elegant solution: is automatic, ensures consistency, and requires less management overhead.

Only problem right now is that we don't have a great framework for how to handle the strategist fee on delegated strats.  

What do you think?

Reply:
We don't have *any* framework for handling these delegated fund strategies currently. We add the mgmt fee fix as a "user benefit" since we see the mgmt fee as being core profit for our Vaults. The performance fee sort of pays for overhead costs, loosely, so that's why we didn't make the change to the performance fee. The strategists also still put in effort to develop the strategy, which is why we didn't adjust that fee either.

Reply:
Sure. I agree. Fees overall need a fresh look. Either we stay true to 2/20 or we adjust to a new scheme that makes more sense. 

For now, with yields so low, i worry that fees in excess of 2/20 will result in extremely low (or zero) yield to user, harming our TVL. all Maker strats right now are at or near net 0 yield for user.

Reply:
Have we considered... disabling the Maker strats?

Reply:
as long as they're earning revenue for us and not net negative, we should keep them enabled.